### PR TITLE
Use LWP to send https request in bmcdiscover instead of curl command

### PR DESCRIPTION
**Pull request by xuweibj**
#3100 
Use ``LWP`` to send login and 'GET' request to avoid use ``curl`` command in bmcdiscover.pm

### CODE REVIEW

File: xCAT-server/lib/xcat/plugins/bmcdiscover.pm [hunk 1]
```diff
@@ -1040,14 +1043,20 @@ sub bmcdiscovery_openbmc{
...
+    my $response = $brower->request($request);
     
-    if ($output =~ /\"status\": \"ok\"/) {
-        my $req_output = `curl -b $cjar_file -k https://$ip/xyz/openbmc_project/inventory/system/chassis/motherboard`;
-        my $response = decode_json $req_output;
+    if ($response->is_success) {
+        my $req_url = "https://$ip/xyz/openbmc_project/inventory/system/chassis/motherboard";
+        my $get_req = HTTP::Request->new('GET', $req_url, $header);
+        my $req_output = $brower->request($get_req);
+        my $response = decode_json $req_output->content;
```
Comment (chenglch) [inline xCAT-server/lib/xcat/plugins/bmcdiscover.pm]: Why don't you check the request result like line 1055?

File: xCAT-server/lib/xcat/plugins/bmcdiscover.pm [hunk 2]
```diff
@@ -1080,10 +1089,8 @@ sub bmcdiscovery_openbmc{
             $node =~ s/(.*)/\L$1/g;
             $node =~ s/[\s:\._]/-/g;
         }
-
-        unlink $cjar_file;
     } else {
-        if ($output =~ /\"description\": \"Invalid username or password\"/) {
+        if ($response->content =~ /\"description\": \"Invalid username or password\"/) {
             xCAT::MsgUtils->message("W", { data => ["Invalid username or password for $ip"] }, $::CALLBACK); 
         }
```
Comment (chenglch) [inline xCAT-server/lib/xcat/plugins/bmcdiscover.pm]: As you use the library now, if there is unknow failure( maybe 404, 503), you should keep some log or send some message back to the user, return immediately may not be a good choice.

File: xCAT-server/lib/xcat/plugins/bmcdiscover.pm [hunk 3]
```diff
@@ -1040,14 +1043,20 @@ sub bmcdiscovery_openbmc{
...
+    my $request = HTTP::Request->new( 'POST', $url, $header, $data );
+    my $response = $brower->request($request);
     
-    if ($output =~ /\"status\": \"ok\"/) {
-        my $req_output = `curl -b $cjar_file -k https://$ip/xyz/openbmc_project/inventory/system/chassis/motherboard`;
-        my $response = decode_json $req_output;
+    if ($response->is_success) {
+        my $req_url = "https://$ip/xyz/openbmc_project/inventory/system/chassis/motherboard";
+        my $get_req = HTTP::Request->new('GET', $req_url, $header);
+        my $req_output = $brower->request($get_req);
```
Comment (chenglch) [inline xCAT-server/lib/xcat/plugins/bmcdiscover.pm]: Suggest considering a proper name for these variables, now there are $request, $get_req, $req_output and $response.

File: xCAT-server/lib/xcat/plugins/bmcdiscover.pm (hunk 2 above)
Comment (xuweibj) [inline xCAT-server/lib/xcat/plugins/bmcdiscover.pm]: If it returns other error code, we did not consider it as a bmc, so will not print message.

**Pull request merged**